File: SecondYearCS/OOP/Civitas/src/civitas/SorpresaPorJugador.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorJugador extends Sorpresa{
     
    private float valor; 
    
    
    //Resto de sorpresasa
    SorpresaPorJugador(float valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    /**
     * @brief Todos los jugadores dan dinero al jugador actual
     */
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if( this.jugadorCorrecto(actual, todos) ){
            informe(actual,todos);
            for(int i=0;i<todos.size();i++){
                if(i==actual)
                    todos.get(i).modificarSaldo(valor*(-1)* (todos.size()-1) );
                else
                    todos.get(i).modificarSaldo(valor);
            }
        }
    }

    @Override
    public String toString(){
       return super.toString() + " valor: " + valor;
    }

}

File: SecondYearCS/OOP/CivitasRB/lib/SorpresaPorJugador.rb
require_relative 'sorpresa.rb'
module CivitasRB
  class SorpresaPorJugador < Sorpresa
    attr_accessor  :valor
    #Constructores por defecto y con parametro asdfas
    def initialize(valor,texto)
      super(texto)
      @valor = valor
    end

    def aplicarAJugador(act,todos)
      if(self.jugadorCorrecto(act, todos))
        self.informe(act,todos)
        todos.each_with_index do |jugador,index|
          if index==act
            jugador.modificarSaldo(@valor*(-1)*(todos.length-1))
          else
            jugador.modificarSaldo(@valor)
          end     
        end
      end
    end

   def to_s
      return super.to_s + " valor: " + @valor  +  " "
    end
    
  end
end